ring/src/lib.rs

```diff
@@ -35,21 +38,36 @@ pub fn find_complement_point<Curve: SWCurveConfig>() -> Affine<Curve> {
     }
 }
 
+pub fn find_random_point<F: PrimeField, P: AffineRepr<BaseField = F>>() -> P {
+    let mut x: u8 = 0;
+    loop {
+        let p = P::from_random_bytes(&[x]);
+        if p.is_some() && !p.unwrap().is_zero() {
+            // && !p.unwrap().is_in_correct_subgroup_assuming_on_curve() {
+            return p.unwrap().clear_cofactor();
+        }
+        x = x + 1;
+    }
+}
+
 // Try and increment hash to curve.
-pub(crate) fn hash_to_curve<F: PrimeField, Curve: SWCurveConfig<BaseField = F>>(message: &[u8]) -> Affine<Curve> {
+pub(crate) fn hash_to_curve<F: PrimeField, P: AffineRepr<BaseField = F>>(message: &[u8]) -> P {
     use blake2::Digest;
     let mut seed = message.to_vec();
     let cnt_offset = seed.len();
+    let mut no_tries: usize = 0;
+
     seed.push(0);
     loop {
         let hash: [u8; 64] = blake2::Blake2b::digest(&seed[..]).into();
-        let x = F::from_le_bytes_mod_order(&hash);
-        if let Some(point) = Affine::<Curve>::get_point_from_x_unchecked(x, false) {
+        if let Some(point) = P::from_random_bytes(&hash) {
             let point = point.clear_cofactor();
-            assert!(point.is_in_correct_subgroup_assuming_on_curve());
-            return point
+            //assert!(point.is_in_correct_subgroup_assuming_on_curve());
```

Comment:
```suggestion
```
I guess this is a safe assumption as we're clearing the cofactor.
Do you think we also need to check if we hit the identity here? (i.e. !is_zero)

Reply:
Good Point! I'll add the check as well.